types/backbone/index.d.ts

```diff
@@ -568,26 +568,28 @@ declare namespace Backbone {
 
         model: TModel;
         collection: Collection<TModel>;
-        // template: (json, options?) => string;
-        setElement(element: HTMLElement | JQuery, delegate?: boolean): View<TModel>;
-        setElement(element: any): View<TModel>;
-        id: string;
+        setElement(element: HTMLElement | JQuery): this;
+        id?: string;
         cid: string;
-        className: string;
+        className?: string;
         tagName: string;
```

Comment:
Why make `id` and `className` optional but not `tagName`?

Reply:
`tagName` has `div` default value: https://github.com/jashkenas/backbone/blob/master/backbone.js#L1313
`model` and `collection` are also optional, but changing them to optional attributes breaks a lot of code.